NIC94 Post (#1467)
GoogleChrome/developer.chrome.com: site/en/blog/new-in-chrome-94/index.md

---
title: New in Chrome 94
description: >
  Chrome 94 is rolling out now! The default color space for `<canvas>`

>   elements is now formally defined as sRBG, and you can change it to Display

Reviewer: I think this should be "sRGB", here and below.

Author: Whoops, correct!

  origin trial. And, there's plenty more!
layout: 'layouts/blog-post.njk'
date: 2021-09-21
authors:
  - petelepage
hero: 'image/0g2WvpbGRGdVs0aAPc6ObG7gkud2/bFEcnAWdzNkXaSSp0hX4.png'
alt: >
  New in Chrome hero logo
tags:
  - new-in-chrome
  - chrome-94
---

{% YouTube id='c5XIDt90VgY' %}

Here's what you need to know:

* The [default color space](#canvas-colorspace) for `<canvas>` elements is
  now formally defined  in the spec as *sRBG*, and you can change it to
  *Display P3*.
* [WebCodecs](#webcodecs) is a new, low level way to access built in audio and
  video codecs, important for streaming games, video editors, and such.
* [WebGPU](#webgpu) starts its origin trial.
* The [PWA Summit](#pwa-summit) is coming up October 6-7.
* And there's plenty [more](#more).

I'm [Pete LePage](https://petelepage.com), working, and shooting
from home, let's dive in and see what's new for developers in Chrome 94.

## Default color space for `canvas` elements {: #canvas-colorspace }

How color is rendered on screen is critical to some users. For photographers,
print illustrators, and many others, the colors on screen need to match what's

> printed. Starting in [Chrome 94][cs-canvas], `<canvas>` elements are fully

Reviewer 2: It seems like the link should be on "`<canvas>` elements are fully color managed".

color managed using *sRGB*. Previously, *sRGB* was convention, but not
explicitly defined in the spec.

```js
opts = {colorSpace:'display-p3'};
const ctx = canvas.getContext('2d', opts);
```

> More importantly - you can now specify which color space to use when creating

Reviewer 2: ```suggestion
More importantly, you can now specify which color space to use when creating
```

a `<canvas>` rendering context 2d object, or an `ImageData` object, including
the P3 color space.

## WebCodecs {: #webcodecs }

Putting video on a page is easy enough. But, if you need to do something a
little more complex, and interact with the components of a video stream,

> it's hard! And, typically requires you to use Web Assembly to ship

Reviewer 2: ```suggestion
it's hard, and typically requires you to use Web Assembly to ship
```

Reviewer 2: The other option was to add a verb to the second sentence ("And *it* typically...). Since this is a transcript...

[your own codecs][wasm-av1]!

> But shipping your own codec, means writing code the browser already has, and

Reviewer 2: ```suggestion
But shipping your own codec means writing code the browser already has, and
```

it can't take advantage of hardware acceleration! The Web Codecs API makes it
possible to use the media components and codecs that are already in the browser.

Personally, I've always struggled to remember the correct command line switches
to encode a video for the web, or to convert a GIF to a video file. Using
the APIs available via WebCodecs, I could quickly build a web app that
reads a file and exports the correct files needed for the web.

<figure class="w-figure">
  {% Img src="image/sQ51XsLqKMgSQMCZjIN0B7hlBO02/lEovMYp8oh1JSClCLCiD.png", alt="The path from a Canvas or an ImageBitmap to the network or to storage", width="800", height="393" %}
  <figcaption class="w-figcaption">
    The path from a <code>Canvas</code> or an <code>ImageBitmap</code> to the
    network or to storage
  </figcaption>
</figure>

Web apps that require full control over the way media content is processed,

> like video editors, video conferencing, streaming apps, and so on, access to

Reviewer 2: ```suggestion
like video editors, video conferencing, streaming apps, and so on. Access to
```

> the browsers built in media controls, this is huge.

Reviewer 2: Is this what you actually said in the video? It looks like it should be this:

*For* web apps that require full control over the way media content is processed, like video editors, video conferencing, streaming apps, and so on, access to the browsers built in media controls~~, this~~ is huge.

(Double tildes are supposed to be strike through, but GitHub doesn't support them.)

> Showing anything useful in 30 seconds is hard, so checkout

Reviewer 2: ```suggestion
Showing anything useful in thirty seconds is hard, so check out
```

lots of code and a few cool demos!

## WebGPU {: #webgpu }

WebGPU is a new API that exposes modern graphics capabilities, specifically
Direct3D 12, Metal, and Vulkan. You can think of it like WebGL, but it
provides access to more advanced features of the GPU, and it also provides
support for performing [general computations on the GPU][gpu-compute].

<figure class="w-figure">
  {% Img src="image/vvhSqZboQoZZN9wBvoXq72wzGAf1/WHoJmX2IU7roV4iabH6M.png", alt="Architecture diagram showing WebGPUs connection between OS APIs and Direct3D 12, Metal, and Vulkan.", width="800", height="313" %}
  <figcaption class="w-figcaption">
    WebGPU architecture diagram
  </figcaption>
</figure>

It's starting an [origin trial][ot-gpu] in Chrome 94, and both Safari and
Firefox are currently working on their implementations.

<figure class="w-figure">
  {% Video src="video/vvhSqZboQoZZN9wBvoXq72wzGAf1/Xb7LvsJ5e8efTssp94c6.mov", autoplay=true, muted=true, playsinline=true, loop=true %}
  <figcaption class="w-figcaption">
    A Babylon.js demo of a rough sea being simulated using WebGPU's compute
    shader capability.
  </figcaption>
</figure>

> Francois has a great article [Access modern GPU features with WebGPU][wd-gpu]

Reviewer 2: ```suggestion
François has a great article, [Access modern GPU features with WebGPU][wd-gpu]
```

on web.dev with the details, and compares the performance of matrix

> multiplication running on the CPU vs the GPU. Here's a hint, the

Reviewer 2: ```suggestion
multiplication running on the CPU versus the GPU. Here's a hint. The
```

planning your migration off of it.

And the [virtual keyboard API][wd-vkeyboard] gives you more control over how
and when the virtual, on-screen keyboard is shown. It allows you to explicitly
control any scrolling behaviour, or changes to layout when the keyboard
appears or disappears.

## Further reading

This covers only some of the key highlights. Check the links below for
additional changes in Chrome 94.

* [What's new in Chrome DevTools (94)](/blog/new-in-devtools-94/)
* [Chrome 94 deprecations & removals](/blog/deps-rems-94/)
* [ChromeStatus.com updates for Chrome 94](https://www.chromestatus.com/features#milestone%3D94)
* [What's new in JavaScript in Chrome 94](https://v8.dev/blog/v8-release-94)
* [Chromium source repository change list](https://chromium.googlesource.com/chromium/src/+log/93.0.4577.69..94.0.4606.56)
* [Chrome release calendar](https://chromiumdash.appspot.com/schedule)

## Subscribe

To stay up to date, [subscribe](https://goo.gl/6FP1a5)
to [Chrome Developers YouTube channel](https://www.youtube.com/user/ChromeDevelopers/),
and you'll get an email notification whenever we launch a new video.

> I'm Pete LePage, and as soon as Chrome 95 is released, I'll be right here to

Reviewer: did you want to add a personal blog link here as well? :)

Author: Nah - thats fine.
